src/core/algorithm/graphToMarkovChain.test.js

```diff
@@ -203,6 +228,33 @@ describe("core/algorithm/graphToMarkovChain", () => {
       expect(normalize(osmc)).toEqual(normalize(expected));
     });
 
+    it("works on a simple chain with a loop edge", () => {
+      // The loop edge will implicitly correspond to one forward edge with weight 2, and one backward edge with weight 1
+      const loop = edge("loop", n1, n1);
+      const e1 = edge("e1", n1, n2);
+
+      const g = new Graph().addNode(n1).addNode(n2).addEdge(e1).addEdge(loop);
+      const edgeWeight = () => ({forwards: 2, backwards: 1});
+      const osmc = createOrderedSparseMarkovChain(
+        createConnections(g, edgeWeight, 0.0)
+      );
+      const expected = {
+        nodeOrder: [n1.address, n2.address],
+        chain: [
+          {
+            neighbor: new Uint32Array([0, 1]),
+            // Reproduce floating point imprecision
+            weight: new Float64Array([2 / 5 + 1 / 5, 1]),
+          },
+          {
+            neighbor: new Uint32Array([0, 1]),
+            weight: new Float64Array([2 / 5, 0]),
```

Comment:
These would be easier to read, if we had the same comment at the top about expected normalization.
```
      // Total out-weights (for normalization factors):
      //   - for `n1` ... = 5
      //   - for `n2` ... = 1?
```

Reply:
+1